### DISCUSSION TITLE
Audio not working properly

### DISCUSSION THREAD

**Discussion by chicco-carone** [ppy/osu#26765] [Q&A]
### Type

Other

### Bug description

I'm using an unusual setup with my os running in a vm in proxmox with my gpu and usb passed though. 
System info from lutris:
[System]
OS: Arch Linux rolling n/a
Arch: x86_64
Kernel: 6.7.1-zen1-1-zen
Desktop: KDE
Display Server: x11
[CPU]
Vendor: GenuineIntel
Model: Intel(R) Xeon(R) CPU E5-2665 0 @ 2.40GHz
Physical cores: 16
Logical cores: 16
[Memory]
RAM: 32.7 GB
Swap: 4.0 GB
[Graphics]
Vendor: NVIDIA Corporation
OpenGL Renderer: NVIDIA GeForce RTX 2060/PCIe/SSE2
OpenGL Version: 4.6.0 NVIDIA 545.29.06
OpenGL Core: 4.6.0 NVIDIA 545.29.06
OpenGL ES: OpenGL ES 3.2 NVIDIA 545.29.06
Vulkan Version: 1.3.276
Vulkan Drivers: NVIDIA GeForce RTX 2060 (1.3.260)

I have osu installed from the aur package osu-lazer-bin and i use pipewire as my audio server with pipewire alsa installed. I run the game with this variable exported at launch (set in kde options for the game) PIPEWIRE_LATENCY=256/48000. In the audio settings if i use pipewire sound server (which i use on my other pc without issues) the audio starts crackling, so i tried with pulseaudio sound server which didn't crackle but had 100/200 ms of delay, so in the end i tried to select directly my audio card and after i did it the audio played fine and the latency was normal but as soon as i opened any other application that uses audio in the system (both voip clients and regular audio) i couldn't hear anything from the other applications since osu was connecting directly to alsa and pipewire didn't have anywere to reproduce the audio. If i start playing audio before opening the game and i open it the audio starts crackling and i get the error `Bass faulted with error code 46: busy`

### Version

2024.114.0-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14077627/compressed-logs.zip)

**Comment by AeroBliss** [comment]
Try the AppImage and the Flatpak. Bass is the proprietary audio framework osu runs. They have a forum on their website too.

**Reply by chicco-carone** [reply]
The osu-lazer-bin package Just repackages the appimage into the system so it's like I'm using the appimage

**Comment by RambilE** [comment]
is there a solution? audio worked for me in lazer, but recently it just gives error 46 and 3, sometimes creating two audio channels without sound

**Comment by private04** [comment]
Hello, I run into the same problem a couple of days ago, for me changing the audio device from my headphones to the Pulseaudio server fixed it and allowed me to run multiple apps with audio without any issue.

I tried pipewire server but the audio was glitched and the app was sped up (no idea how that happened). If you're using pipewire, make sure you have installed pipewire-pulse in your system. [https://wiki.archlinux.org/title/PipeWire](url). I believe KDE has it installed by default.

**Reply by Antonio225t** [reply]
This worked for me actually. Go to Osu! main menu, then `settings` → `Audio` and set `Device output` to **PulseAudio Sound Server**. Or try a different server like **PipeWire Sound Server** (if it works) or **Open Sound System**.

If all of the above do not work for you try playing with the available options until (hopefully) you find the one suitable for you.
Take care :+1:

**Comment by yippii** [comment]
@chicco-carone Maybe you should go check your pipewire settings to see if you had restricted the quantum in anyway, as osu! defaults to use 64 as the quantum value so there would be less latency (ofc for a rhythm game), which if you restricted it, it would produce unexpected results such as audio crackling, I solved this in NixOS by changing the min-quantum from 2048 to 64, and default-quantum to 256, which solved it.

**Reply by rszyma** [reply]
> osu! defaults to use 64 as the quantum value

Not true by the way. The default latency in osu is 10ms, which is 441 quantum (speaking in pipewire terms) at 44.1 khz.

Why you see 64 as default is purely because it's forced on osu by your pipewire config. Or less likely, but also possible, that some other app running PIPEWIRE_QUANTUM env var and forces all audio apps for as long as that app that has set it is running.
